### PR TITLE
Change tileset additional changes

### PR DESCRIPTION
**Pull request by SJuliez**
Originally a fix for the hex shifting error but this contains a few additional changes concerning events and moves the "change tileset" function to the Board.

### CODE REVIEW

**Comment by SJuliez** [comment]
Not sure what to make of the board_validator....

**Comment by HammerGS** [comment]
Arlith added the Board Validator a few years ago, to prevent crashing and find and fix some of the handmade boards. I use it a lot when adding maps from mapmakers I don't now.

**Comment by SJuliez** [comment]
Well, I didn't change any of the boards...
Anyway, why does it fail now? Instead of yesterday...

**Comment by Windchild292** [comment]
It was just added yesterday

**Comment by sixlettervariables** [comment]
The new board validation check runs whenever a change is made which could impact loading boards, including code which reads boards.

File: megamek/src/megamek/client/ui/swing/TilesetManager.java [hunk 1]
```diff
@@ -126,12 +128,18 @@
...
+    
+    // The TileManager is associated with a game passed down from the Boardview
+    // and must pass this on to the hextileset for it to register listeners
+    private IGame game;
 
     /**
      * Creates new TilesetManager
      */
-    public TilesetManager(BoardView1 bv) throws IOException {
+    public TilesetManager(BoardView1 bv, IGame g) throws IOException {
```
Comment (NickAragua) [inline megamek/src/megamek/client/ui/swing/TilesetManager.java:139]: I'd like to point out that BoardView already has a game object, so there's not a need to pass it around.

Comment (SJuliez) [reply megamek/src/megamek/client/ui/swing/TilesetManager.java]: OK, I have another solution. If there is some obviously better one, please do point me to it. I'm open towards at least a megamek-internal philosophy on that. I made boardview's game public and final and removed the pass to the Tilesetmanager.

Comment (SJuliez) [reply megamek/src/megamek/client/ui/swing/TilesetManager.java]: Is there something I should do because of the board validator?

Comment (NickAragua) [reply megamek/src/megamek/client/ui/swing/TilesetManager.java]: Yeah, if you click on the details for the Validate Boards check, you can look at which boards fail as a result of the changes made and address those.

**Comment by SJuliez** [comment]
@NickAragua OK, I corrected the boards. The one board reporting as not working is ok but doesnt load because of #1898

**Pull request merged**